Implements phone verification and payment history

tej17584 opened the pull request:
Adds phone verification flow using OTP and payment history screen.

- Implements a phone verification flow using OTP with UI components and logic for sending and verifying codes.
- Introduces a payment history screen to allow users to review their past transactions.

tej17584 committed:
Auto stash before merge of "feature/expo-update" and "develop"

tej17584 committed:
Adds payment history screen

Implements the payment history screen, displaying a list of user payments with details such as status, date, amount, and associated articles.

Includes error handling and empty state management.
Adds PaymentCard component to display individual payment information.

tej17584 committed:
Refactors payment history screen

Improves the payment history screen by introducing a dedicated component for the empty state, enhancing code readability and maintainability.

Moves the empty state logic into a new `EmptyPaymentHistory` component.
Updates the main screen to use this new component, reducing complexity.
Fetches payment history data from backend.
Updates the subtitle in i18n files

tej17584 committed:
Adds payment history feature

Implements the payment history feature, allowing users to view their past transactions.

This includes the UI components for displaying individual payment cards and handling empty state scenarios.
It also introduces a hook for fetching payment history data from the API.

tej17584 committed:
Adds phone verification feature

Implements phone number verification using OTP.

Integrates a phone input component with country code selection.

Adds OTP sending and verification endpoints to the API.

Adds translations for the new feature.

tej17584 committed:
cambios

tej17584 committed:
Uses full phone number for OTP verification.

Ensures that the OTP verification process utilizes the full phone number, including the country code.
This change addresses an issue where the OTP was being sent and verified using only the local phone number, leading to failures in international contexts.
The phone number is now extracted from the input field, with spaces removed, and used in both sending and verifying the OTP.
Updates the API endpoint for OTP to be under the user namespace.

tej17584 committed:
Adds user feedback for phone verification

Improves user experience by displaying informative toast messages for various scenarios in the phone verification process, such as invalid phone numbers, OTP codes, and server errors.
This enhancement provides immediate feedback to the user, guiding them through the verification steps and preventing confusion.

Additionally, it clears the OTP input field upon a failed verification attempt, preventing the user from resubmitting the same incorrect code.

tej17584 committed:
Improves phone verification flow and UI.

Enhances the phone verification process by adding validation to prevent users from verifying already verified numbers.
It also improves the phone input validation and updates the Payments History screen's subtitle to correct a typo.

tej17584 committed:
Adds phone verification flow

Implements a phone verification wizard with OTP, allowing users to verify their phone numbers.

Introduces a new hook for managing the phone verification process and related logic.
Adds a new toast hook for displaying success/error messages during the process.
Adds tests for both the wizard component and the verification hook.

tej17584 committed:
Merge branch 'develop' into feature/verifyPhone-PaymentHistory

totisama commented on app/(tabs)/account/payments-history.tsx:
```diff
@@ -1,26 +1,102 @@
+  }, [refetch]);
+
+  // Loading state
+  if (status === 'loading' || status === 'idle') {
```
use esta constante en vez de la magic string
`REQUEST_STATUS`

totisama committed:
display text to show if phone is verified or not

totisama committed:
use correct text for back button

tej17584 committed:
PR comments on loading and more

tej17584 committed:
Hides loading indicator on manual refresh

Hides the loading indicator when the user manually refreshes the screen.
This prevents the loading indicator from briefly flashing on top of the refreshed content, providing a smoother user experience.

tej17584 committed:
fix: update VerifyPhoneWizard tests for independent loading states

- Replace failing 'should disable buttons when loading' test
- Add new 'should have independent loading states for each button' test
- Fix test to use getByPlaceholderText instead of non-existent testID
- Tests now reflect independent loading states (isSending, isVerifying, isResending)
- All 19 VerifyPhoneWizard tests now passing
